base/pcre.jl

```diff
@@ -100,6 +101,7 @@ end
 function study(regex::Ptr{Void}, options::Integer)
     # NOTE: options should always be zero in current PCRE
     errstr = Array(Ptr{UInt8},1)
+    errstr[1] = C_NULL
```

Comment:
mostly this was caused because I accidentally was passing a copy of the array in the pointer, rather than the array itself (during work on this PR). it shouldn't strictly be necessary on master (since pcre should set this variable)